graphql_compiler/cost_estimation/statistics.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from abc import ABCMeta, abstractmethod
+
+import six
+
+
+@six.python_2_unicode_compatible
+@six.add_metaclass(ABCMeta)
+class Statistics(object):
+    """Abstract class for statistics regarding GraphQL objects and vertex field values.
+
+    For the purposes of cost estimation, we need statistics to provide estimates for costs of
+    certain operations like edge traversal and @filter directive usage.
+    This class provides an interface for these statistics.
+    All statistics except get_class_count are optional, so if the statistic doesn't exist, a
+    value of None should be returned.
+    """
+    def __str__(self):
+        """Return a human-readable unicode representation of the Statistics object."""
+        raise NotImplementedError()
+
+    def __repr__(self):
+        """Return a human-readable str representation of the CompilerEntity object."""
+        return self.__str__()
+
+    @abstractmethod
+    def get_class_count(self, class_name):
+        """Return how many objects have, or inherit, the given class name.
+
+        Args:
+            class_name: str, either a vertex class name or an edge class name defined in the
+                        GraphQL schema.
+
+        Returns:
+            - Number of objects with, or inheriting the given class name
```

Comment:
nit: Please start with the type of the return value first in this docstring and in the ones below.